src/stores/models/Transaction.js

```diff
@@ -0,0 +1,43 @@
+import { Token, TransactionType } from '../../constants';
+import { gasToQtum, satoshiToDecimal } from '../../helpers/utility';
+
+export default class Transaction {
+  type = ''
+  txid = ''
+  status = ''
+  createdTime = ''
+  blockNum = 0
+  blockTime = ''
+  gasLimit = 0
+  gasPrice = 0
+  gasUsed = 0
+  version = 0
+  senderAddress = ''
+  receiverAddress = ''
+  topicAddress = ''
+  oracleAddress = ''
+  name = ''
+  optionIdx = 0
+  token = ''
+  amount = ''
+  topic
+
+  constructor(transaction, app) {
+    Object.assign(this, transaction);
+    this.app = app;
+    this.gasLimit = Number(this.gasLimit);
+    this.gasPrice = Number(this.gasPrice);
+    this.fee = gasToQtum(this.gasUsed);
+
+    if (this.token === Token.Bot) {
+      if (this.type !== TransactionType.ApproveCreateEvent
```

Comment:
nit: `![ApproveCreateEvent, ApproveSetResult, ApproveVote].some(this.type)` basically the logic is "if not 1 of these is true"

Reply:
decided to use `_.includes`